optimizely/entities.py

```diff
@@ -41,18 +57,18 @@ def __init__(self, id, key, experimentIds, **kwargs):
 class Experiment(BaseEntity):
     def __init__(
         self,
-        id,
-        key,
-        status,
-        audienceIds,
-        variations,
-        forcedVariations,
-        trafficAllocation,
-        layerId,
-        audienceConditions=None,
-        groupId=None,
-        groupPolicy=None,
-        **kwargs
+        id: str,
+        key: str,
+        status: str,
+        audienceIds: list[str],
+        variations: list[VariationDict],
```

Comment:
This class is mirroring the names from the datafile, so changing this field name is non trivial. Instead of being able to auto unpack the datafile object, we would have to manually assign in a couple different places. For example:
```diff
 for experiment_dict in layer.experiments:
-     self.experiment_id_map[experiment_dict['id']] = entities.Experiment(**experiment_dict)
+     self.experiment_id_map[experiment_dict['id']] = entities.Experiment(
+                    id=experiment_dict['id'],
+                    key=experiment_dict['key'],
+                    status=experiment_dict['status'],
+                    audienceIds=experiment_dict['audienceIds'],
+                    variationsDict=experiment_dict['variations'],
+                    forcedVariations=experiment_dict['forcedVariations'],
+                    trafficAllocation=experiment_dict['trafficAllocation'],
+                    layerId=experiment_dict['layerId'],
+                    audienceConditions=experiment_dict.get('audienceConditions'),
+                    groupId=cast(Optional[str], experiment_dict.get('groupId')),
+                    groupPolicy=cast(Optional[str], experiment_dict.get('groupPolicy'))
+                )
```

Might not be worth it?